### PR TITLE
fix: return 403 when non-admin tries to do admin requests

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
With this PR, all the graph `admin requests` if done by the `non-admin` users will respond with `403 Forbidden` instead of `401 Unauthorized`.

Motivation: https://github.com/owncloud/ocis/pull/5742#issuecomment-1482547906
>Now the `401 Unauthorized` vs `403 Forbidden` vs `404 Not Found` status code may look tricky, but hara are the guidelines:
>
>- we don't want to expose existence of resources if a user has no access to them, so we return a `404 Not Found` instead of a `403 Forbidden`.
>- when a user has access to a resource but tries to execute an action that he does not have enough permissions for, e.g. when he tries to write to a read only share, we return a `403 Forbidden`

- We are expecting `403` in some test scenarios.

But the above comment doesn't describe more about `401 Unauthorized` vs `403 Forbidden`. It is agreeable to respond with `403` when non-admin users do admin requests but the standard/policy and also from a security point of view, it could be different in oCIS.

The main motive of the PR is to have some conclusion and if agreed upon then hopefully merge the fix, if not then change the test expectation.

\
**Non-admin scenarios: should they respond with `403`, `401` or `404` :question: ** 
```
- try to change other users' role
- try to create a new user
- try to delete (own / non-existing / another / disabled) account
- try to change (own / another) account (email / displayname / password)
- try to disable/enable another user
- try to get all users
- try to get users of a group
- try to get users with a role
- try to add/remove (own/another) account to a group
- try to create a new group
- try to delete/rename a group
- try to get all groups
```

## Related Issue
- Fixes https://github.com/owncloud/ocis/issues/5938

## How Has This Been Tested?
- test environment:

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by micbar** [comment]
@saw-jan Still a thing? or close?

**Comment by saw-jan** [comment]
> @saw-jan Still a thing? or close?

I there can be a clarification on the following query, then it would be helpful to decide whether to work on it or close.
>Non-admin scenarios: should they respond with `403`, `401` or `404` ❓

Sorry, I totally forgot about this PR

File: services/graph/pkg/middleware/requireadmin.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func RequireAdmin(rm *roles.Manager, logger log.Logger) func(next http.Handler)
 				return
 			}
 
-			errorcode.AccessDenied.Render(w, r, http.StatusUnauthorized, "Unauthorized")
+			errorcode.AccessDenied.Render(w, r, http.StatusForbidden, "Forbidden")
```
Comment (saw-jan) [inline services/graph/pkg/middleware/requireadmin.go:51]: Needs to be reviewed by dev team if this is correct and right place to do it

**Comment by saw-jan** [comment]
Failing tests are expected ones

**Comment by phil-davis** [comment]
The other thing that we need to check is that these status codes should not "leak" information about what exists and does not exist.

When a user with not enough privileges tries to modify or delete something (user, group, space...) that exists, they get a 403 "forbidden" - good. If they try the same modify or delete to a non-existent user, group, space, then they should also get a 403 "forbidden". If they get 404 "not found" then they will know that the item really does not exist, and that "403" indicates that the item exists (and they do not have permission).

Similarly for "create" requests - if a 403 "forbidden"is given when trying to create a new user that does not already exist, then 403 should also be given if "create" is attempted for a user that already exists.

**Review by dragonchaser — APPROVED** [review]

**Comment by phil-davis** [comment]
@saw-jan is there an issue about testing the cases where 403 and/or 404 status responses might leak information?
Or do the test scenarios already cover that well?

**Comment by saw-jan** [comment]
> @saw-jan is there an issue about testing the cases where 403 and/or 404 status responses might leak information? Or do the test scenarios already cover that well?

There are some tests like this one
https://github.com/owncloud/ocis/blob/775913c8b5839d4b45c93dbafa55ac6465eea1d3/tests/acceptance/features/apiGraphUserGroup/deleteUser.feature#L78-L82

**Comment by phil-davis** [comment]
> There are some tests like this one

Good. If there are the same sort of test scenarios for non-existent group, non-existent space... for trying to delete and trying to modify then we are good.

**Comment by saw-jan** [comment]
>Good. If there are the same sort of test scenarios for non-existent group, non-existent space... for trying to delete and trying to modify then we are good.

I will check for other test cases

**Pull request merged**